Sensus.iOS.Tools/Concurrent/MainConcurrent.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright 2014 The Rector & Visitors of the University of Virginia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading;
+using Sensus.Tools;
+using Xamarin.Forms;
+
+namespace Sensus.iOS.Concurrent
+{
+    public class MainConcurrent: Disposable, IConcurrent
+    {
+        public void ExecuteThreadSafe(Action action)
+        {
+            if (action == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            var runWait = new ManualResetEvent(false);
+
+            Device.BeginInvokeOnMainThread(() =>
+            {
+                try
+                {
+                    action();
+                }
+                finally
+                {
+                    runWait.Set();
+                }
+            });
+
+            runWait.WaitOne();
```

Comment:
Looks like it. I'd simply copied the original code in RunOnMainThreadNative. I'll fix the issue.